databases.go

```diff
@@ -0,0 +1,221 @@
+	if err != nil {
+		panic(err)
+	}
+	return fmt.Sprintf("%s/engines", endpoint)
+}
+
+func (resp *DatabaseEnginesPagedResponse) appendData(r *DatabaseEnginesPagedResponse) {
+	resp.Data = append(resp.Data, r.Data...)
+}
+
+type DatabaseTypesPagedResponse struct {
+	*PageOptions
+	Data []DatabaseType `json:"data"`
+}
+
+func (DatabaseTypesPagedResponse) endpoint(c *Client) string {
+	endpoint, err := c.Databases.Endpoint()
+	if err != nil {
+		panic(err)
+	}
+	return fmt.Sprintf("%s/types", endpoint)
+}
+
+func (resp *DatabaseTypesPagedResponse) appendData(r *DatabaseTypesPagedResponse) {
+	resp.Data = append(resp.Data, r.Data...)
+}
+
+// A Database is a instance of Linode Managed Databases
+type Database struct {
+	ID              int          `json:"id"`
+	Status          string       `json:"status"`
+	Label           string       `json:"label"`
+	Hosts           DatabaseHost `json:"hosts"`
+	Region          string       `json:"region"`
+	Type            string       `json:"type"`
+	Engine          string       `json:"engine"`
+	Version         string       `json:"version"`
+	ClusterSize     int          `json:"cluster_size"`
+	ReplicationType string       `json:"replication_type"`
+	SSLConnection   bool         `json:"ssl_connection"`
+	Encrypted       bool         `json:"encrypted"`
+	AllowList       []string     `json:"allow_list"`
+	InstanceURI     string       `json:"instance_uri"`
+	Created         *time.Time   `json:"-"`
+	Updated         *time.Time   `json:"-"`
+}
+
+// DatabaseHost for Primary/Secondary of Database
+type DatabaseHost struct {
+	Primary   string `json:"primary"`
+	Secondary string `json:"secondary,omitempty"`
+}
+
+// DatabaseEngine is information about Engines supported by Linode Managed Databases
+type DatabaseEngine struct {
+	ID      string `json:"id"`
+	Engine  string `json:"engine"`
+	Version string `json:"version"`
+}
+
+// DatabaseType is information about the supported Database Types by Linode Managed Databases
+type DatabaseType struct {
+	ID          string            `json:"id"`
+	Label       string            `json:"label"`
+	Class       string            `json:"class"`
+	Deprecated  bool              `json:"deprecated"`
+	VirtualCPUs int               `json:"vcpus"`
+	Disk        int               `json:"disk"`
+	Memory      int               `json:"memory"`
+	ClusterSize []DatabaseCluster `json:"cluster_size"`
+}
+
+// DatabaseCluster Sizes and Prices
+type DatabaseCluster struct {
+	Quantity int          `json:"quantity"`
+	Price    ClusterPrice `json:"price"`
+}
+
+// ClusterPrice for Hourly and Monthly price models
+type ClusterPrice struct {
+	Hourly  float32 `json:"hourly"`
+	Monthly float32 `json:"monthly"`
+}
+
+// DatabaseSSL is the SSL Certificate to access the Linode Managed Database
+type DatabaseSSL struct {
+	Certificate []byte `json:"ca_certificate"`
+}
+
+// DatabaseCredential is the Root Credentials to access the Linode Managed Database
+type DatabaseCredential struct {
```

Comment:
Could we move this to `mysql.go` with the name `MySQLDatabaseCredentials` to future proof for new DB types?